app/src/main/java/org/simple/clinic/drugs/search/DrugsSearchScreen.kt

```diff
@@ -0,0 +1,149 @@
+    DrugsSearchScreen.Key,
+    ScreenDrugsSearchBinding,
+    DrugSearchModel,
+    DrugSearchEvent,
+    DrugSearchEffect>(), DrugSearchUi, UiActions {
+
+  @Inject
+  lateinit var effectHandlerFactory: DrugSearchEffectHandler.Factory
+
+  @Inject
+  lateinit var router: Router
+
+  private val adapter = PagingItemAdapter(
+      diffCallback = DrugSearchListItem.DiffCallback(),
+      bindings = mapOf(
+          R.layout.list_item_drug_search to { layoutInflater, parent ->
+            ListItemDrugSearchBinding.inflate(layoutInflater, parent, false)
+          },
+          R.layout.list_item_drug_search_divider to { layoutInflater, parent ->
+            ListItemDrugSearchDividerBinding.inflate(layoutInflater, parent, false)
+          }
+      )
+  )
+
+  private val drugSearchToolbar
+    get() = binding.drugSearchToolbar
+
+  private val progressIndicator
+    get() = binding.drugSearchProgressIndicator
+
+  private val drugSearchResultsList
+    get() = binding.drugSearchResultsList
+
+  private val searchQueryEditText
+    get() = binding.searchQueryEditText
+
+  override fun defaultModel() = DrugSearchModel.create()
+
+  override fun createUpdate() = DrugSearchUpdate()
+
+  override fun createEffectHandler() = effectHandlerFactory.create(this).build()
+
+  override fun uiRenderer() = DrugSearchUiRenderer(this)
+
+  override fun events() = searchQueryChanges()
+      .compose(ReportAnalyticsEvents())
+      .cast<DrugSearchEvent>()
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    context.injector<Injector>().inject(this)
+  }
+
+  override fun bindView(
+      layoutInflater: LayoutInflater,
+      container: ViewGroup?
+  ) = ScreenDrugsSearchBinding.inflate(layoutInflater, container, false)
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+
+    drugSearchToolbar.setNavigationOnClickListener {
+      router.pop()
+    }
+
+    drugSearchResultsList.adapter = adapter
+    adapter.addLoadStateListener(::drugSearchLoadStateListener)
+
+    searchQueryEditText.showKeyboard()
+  }
+
+  override fun hideSearchResults() {
+    drugSearchResultsList.visibility = View.GONE
+    adapter.submitData(lifecycle, PagingData.empty())
+  }
+
+  override fun showSearchResults() {
+    drugSearchResultsList.visibility = View.VISIBLE
+  }
+
+  override fun setDrugSearchResults(searchResults: PagingData<Drug>) {
+    drugSearchResultsList.scrollToPosition(0)
+    adapter.submitData(lifecycle, DrugSearchListItem.from(searchResults, searchQuery))
+  }
+
+  private fun drugSearchLoadStateListener(combinedLoadStates: CombinedLoadStates) {
+    val isLoading = combinedLoadStates.refresh is LoadState.Loading
+    progressIndicator.visibleOrGone(isLoading)
+  }
+
+  private fun searchQueryChanges(): Observable<UiEvent> {
+    return searchQueryEditText
+        .textChanges()
+        .skipInitialValue()
+        .debounce(500, TimeUnit.MILLISECONDS)
```

Comment:
Hmm, what do you think of adding a `.debounce(java.time.Duration)` extension to make this kind of thing more readable?

Reply:
Sure, personally I think this is more readable. But using `Duration` will be consistent with other usages.

Reply:
More than readable, it is also safer since changing the time unit to something like `SECONDS` without changing the actual value will cause issues.